File: backend/Board/src/main/java/Backend/Board/dto/TaskDTO.java
package Backend.Board.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TaskDTO {
    private Long id;
    
    @NotBlank(message = "Task title is required")
    @Size(min = 1, max = 200, message = "Task title must be between 1 and 200 characters")
    private String title;
    
    @Size(max = 1000, message = "Task description must not exceed 1000 characters")
    private String description;
    
    private List<CommentDTO> comments;
    private UserDTO createdBy;
    private UserDTO assignee;
}

File: frontend/kanban-board-client-app/src/app/core/models/classes/TaskDTO.ts
import { TaskDTO as TaskDTOInterface } from '../requestModels/model/taskDTO';
import { UserDTO } from './UserDTO';
import { CommentDTO } from './CommentDTO';

export class TaskDTO implements TaskDTOInterface {
  id?: number;
  title: string;
  description?: string;
  comments?: Array<CommentDTO>;
  createdBy?: UserDTO;
  assignee?: UserDTO;

  constructor(data: TaskDTOInterface) {
    this.id = data.id;
    this.title = data.title;
    this.description = data.description;
    this.comments = data.comments;
    this.createdBy = data.createdBy;
    this.assignee = data.assignee;
  }
}
